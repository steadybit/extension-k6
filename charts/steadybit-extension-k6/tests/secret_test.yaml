templates:
  - secret.yaml
tests:
  - it: no secret with existing secret
    set:
      k6:
        cloudApiKey: null
        existingSecret: existing-secret
    asserts:
      - hasDocuments:
          count: 0
  - it: manifest should match snapshot with api key
    set:
      k6:
        cloudApiKey: 111-222-333
        existingSecret: null
    asserts:
      - matchSnapshot: { }
  - it: no secret without api key
    set:
      k6:
        cloudApiKey: null
        existingSecret: null
    asserts:
      - hasDocuments:
          count: 0
  - it: should render nothing with default values
    asserts:
      - hasDocuments:
          count: 0

