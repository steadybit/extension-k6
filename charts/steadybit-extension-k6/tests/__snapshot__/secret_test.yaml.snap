manifest should match snapshot with api key:
  1: |
    apiVersion: v1
    data:
      cloud-api-key: MTExLTIyMi0zMzM=
    kind: Secret
    metadata:
      labels: null
      name: steadybit-extension-k6
      namespace: NAMESPACE
    type: Opaque
